optimum/exporters/tflite/convert.py

```diff
@@ -132,10 +138,180 @@ def validate_model_outputs(
         )
 
 
+def create_representative_dataset(signatures, dataset: "Dataset"):
+    def representative_dataset():
+        for sig_name, tf_function in signatures.items():
+            inputs_to_keep = None
+            for example in dataset:
+                if inputs_to_keep is None:
+                    args, kwargs = tf_function.structured_input_signature
+                    args_to_keep = {input_.name for input_ in args if input_.name in example}
+                    kwargs_to_keep = {input_.name for input_ in kwargs.values() if input_.name in example}
+                    inputs_to_keep = args_to_keep | kwargs_to_keep
+                yield sig_name, {name: value for name, value in example.items() if name in inputs_to_keep}
+
+    return representative_dataset
+
+
+def prepare_converter_for_quantization(
+    model: "TFPreTrainedModel",
+    config: "TFLiteConfig",
+    preprocessor: Optional[Preprocessor],
+    signatures: Dict[str, Callable],
+    quantization_config: "QuantizationConfig",
+    converter: "tf.lite.TFLiteConverter",
+):
+    import tensorflow as tf
+
+    if (
+        not config.supports_quantization_approach(quantization_config.approach)
+        and not quantization_config.fallback_to_float
+    ):
+        raise QuantizationApproachNotSupported(
+            f"{model.config.model_type} do not support full {quantization_config.approach} quantization, use "
+            "fallback_to_float=True to fallback to the float implementation for the unsupported ops."
+        )
+
+    str_to_dtype = {"int8": tf.int8, "uint8": tf.uint8}
+    if quantization_config.approach in [QuantizationApproach.INT8, QuantizationApproach.INT8x16]:
+        if preprocessor is None:
+            raise ValueError(
+                "A preprocessor must be passed for INT8 and INT8x16 quantization since it is needed to preprocess "
+                "the calibration dataset."
+            )
+
+        converter.optimizations = [tf.lite.Optimize.DEFAULT]
+
+        # Handling the calibration dataset:
+        # - Either loading the default dataset if no calibration dataset was provided or the required dataset,
+        # - Splitting the dataset with the provided a dataset split or with the first split if none is provided.
+        # - Shuffling the split.
+        # - Selecting num_calibration_samples in the dataset split.
+        # - Batching the dataset.
+        # - Converting it to the TensorFlow format.
+        from ...exporters import TasksManager
+
+        task = TasksManager.get_task_from_model(model)
```

Comment:
This also solves the issue with `text-classification` vs `sequence-classification`.
Again, I do not see why making it explicit is better in any way. We should be able to infer it since it is an auto-class.

Reply:
The task is already known in the export script - here we reinfer the task, and you have to be confident in that `get_task_from_model` does its job and yields the right and task as the one you got during the export.

It's not a big deal but just from design aesthetic pov, we do twice the same thing in two different methods and just hope they are both well implemented - instead of hopping one is well implemented, which is easier to maintain.

But this is just taste, it's fine to me as is.